thrift/lib/cpp2/CMakeLists.txt

```diff
@@ -208,6 +208,7 @@ add_dependencies(
   thriftcpp2
   rpcmetadata
   thriftmetadata
+  RocketUpgrade-cpp2-target
```

Comment:
Would this mean that `${RocketUpgrade-cpp2-SOURCES}` (line 205) becomes outdated?

Reply:
No, both are needed.

Adding `RocketUpgrade-cpp2-target` here means that the RocketUpgrade source files are generated *before* the `thriftcpp2` library is built. This is required since a non-generated source, `HeaderClientChannel.cpp`, depends on a generated header, `RocketUpgradeAsyncClient.h`. If it were to be compiled before the header is generated, you'd get a missing include error.

Adding `${RocketUpgrade-cpp2-SOURCES}` to the `thriftcpp2` sources is needed so that the generated sources are built and linked into the `thriftcpp2` library. If it were omitted you'd get linker errors for undefined symbols.

This also matches the other libraries here. For example, `thriftmetadata` correctly lists `${metadata-cpp2-SOURCES}` in the library sources, but also adds a dependency on `metadata-cpp2-target`.

Looks like the bug was introduced in d94e1037. Previously, there was a separate `rocketupgrade` library which had the appropriate `RocketUpgrade-cpp2-target` dependency. When this was merged into thriftcpp2, this dependency was not carried over.